### PR TITLE
feat(components/atom/backToTop): remove unecesary props, add more, fix behavior on ref targetted win

### PR DESCRIPTION
**Pull request by andresin87**
## ATOM/BACKTOTOP
**TASK**: 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [ ] Refactor

### Description, Motivation and Context

- forwardRef
- Controlled n uncontrolled visibility state management
- targetRef opens a portal to a selected dom element
- fixes the default document targetRef element behavior
- improves performance removing the recursive callback statement for scrolling (setInterval 50ms) and uses the vanillaJS `element.scrollTo()` method.

### Screenshots - Animations

[https://sui-components-dfkotduk9-schibstedspain.vercel.app/workbench/atom/backToTop/demo](https://sui-components-dfkotduk9-schibstedspain.vercel.app/workbench/atom/backToTop/demo)

### BREAKING CHANGES:

remove unecesary props, add more, fix behavior on ref targetted window document and also improves performance (removing recursive setInterval)

### CODE REVIEW

File: components/atom/backToTop/src/index.js [hunk 1]
```diff
@@ -10,109 +15,108 @@ const CLASS_SHOW = `${BASE_CLASS}--show`
...
-    const {scrollTop} = container
-    const {current: intervalId} = intervalRef
-
-    if (scrollTop === 0) clearInterval(intervalId)
-    if (scrollTop) container.scrollTop = scrollTop - scrollSteps
-  }
-
-  const scrollToTop = () => {
-    const intervalId = setInterval(scrollStep, scrollIntervalTime)
-    intervalRef.current = intervalId
```
Comment (nistadev) [inline components/atom/backToTop/src/index.js:65]: 👏🏽👏🏽👏🏽

File: components/atom/backToTop/src/calcBackToTopEngine.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const isDocument = isDocumentElement(target)
+  const scrollTop = isDocument ? window.scrollY : element.scrollTop
+  const clientHeight = isDocument
+    ? window.innerHeight - (element.offsetHeight - element.clientHeight)
+    : element.clientHeight
+  const offsetHeight = isDocument ? window.innerHeight : element.offsetHeight
+  const scrollHeight = isDocument ? element.scrollHeight : element.scrollHeight
+  const borderHeight = offsetHeight - clientHeight
+  properties = {
+    ...properties,
```
Comment (nistadev) [inline components/atom/backToTop/src/calcBackToTopEngine.js]: Variable `properties` is spreaded and reassigned but there are no other properties inside. Why not create it directly? @andresin87

**Review by jordevo — APPROVED** [review]

File: components/atom/backToTop/src/index.js (hunk 1 above)
Comment (andresin87) [reply components/atom/backToTop/src/index.js:65]: 😋

File: components/atom/backToTop/src/calcBackToTopEngine.js (hunk 2 above)
Comment (andresin87) [reply components/atom/backToTop/src/calcBackToTopEngine.js]: ✅

**Review by nistadev — APPROVED** [review]

**Pull request merged**